src/test/java/bio/terra/service/snapshot/SnapshotServiceTest.java

```diff
@@ -165,20 +175,22 @@ class SnapshotServiceTest {
   @BeforeEach
   void beforeEach() {
     service =
-        new SnapshotService(
-            jobService,
-            datasetService,
-            mock(FireStoreDependencyDao.class),
-            mock(BigQuerySnapshotPdao.class),
-            snapshotDao,
-            snapshotTableDao,
-            metadataDataAccessUtils,
-            iamService,
-            ecmService,
-            azureSynapsePdao,
-            rawlsService,
-            duosClient,
-            mock(SnapshotBuilderSettingsDao.class));
+        spy(
```

Comment:
I made this a spy for one test: testCreateSnapshotWithByRequestId() but I feel like there is probably a better way to do test this